File: src/main/java/com/app/service/CommonService.java
package com.app.service;

import com.app.dto.AccountCategoryDto;
import com.app.dto.CategoryDto;
import com.app.dto.LargeCategoryDto;

import java.util.List;

public interface CommonService {
    /**
     * 대분류 카테고리 목록 조회
     * @param categoryType
     * @return List<LargeCategoryDto>
     */
    List<LargeCategoryDto> selectLargeCtgryList(String categoryType);
    /**
     * 카테고리 목록 조회
     * @param categoryType
     * @return CategoryDto
     */
    CategoryDto selectCategoryList(String categoryType);
    /**
     * 통장 카테고리 목록 조회
     * @param accountCategoryType
     * @return List<AccountCategoryDto>
     */
    List<AccountCategoryDto> selectAccountCategoryList(String accountCategoryType);
}

File: frontend/src/service/common_service.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/common/';

class CommonService {
    /**
     * 대분류 카테고리 목록 조회
     * @param {*} categoryType 
     * @returns 
     */
    selectLargeCtgryList(categoryType) {
        console.log('categoryType', categoryType);
        return axios
            .get(API_URL + 'large-category/list', { params: { categoryType: categoryType } })
            .then(response => {
                console.log('response', response);
                return response.data;
            });
    }
    /**
     * 카테고리 목록 조회
     * @param {*} categoryType
     * @returns
     */
    selectCategoryList(categoryType) {
        console.log('categoryType', categoryType);
        return axios
            .get(API_URL + 'category/list', { params: { categoryType: categoryType } })
            .then(response => {
                console.log('response', response);
                return response.data;
            });
    }
    /**
     * 통장 카테고리 목록 조회
     * @param {*} accountCategoryType
     * @returns
     */
    selectAccountCtgryList(accountCategoryType) {
        console.log('accountCategoryType', accountCategoryType);
        return axios
            .get(API_URL + 'account-category/list', { params: { accountCategoryType: accountCategoryType } })
            .then(response => {
                console.log('response', response);
                return response.data;
            });
    }

}


export default new CommonService();